### PR TITLE
aspects: support unmatched placeholders w/ unset

### PR DESCRIPTION
**Pull request by MiguelPires**
Since the unset logic has diverged more significantly from the set logic lately, the first commit splits them into specific methods with no other changes. The second commit adds logic to fulfil

### CODE REVIEW

File: aspects/aspects.go [hunk 1]
```diff
@@ -406,12 +407,16 @@ type expandedMatch struct {
...
 
-// Set sets the named aspect to a specified value.
+// Set sets the named aspect to a specified non-nil value.
 func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
 	if err := validateAspectDottedPath(request, nil); err != nil {
 		return badRequestErrorFrom(a, "set", request, err.Error())
 	}
 
+	if value == nil {
+		return fmt.Errorf("Set value cannot be nil")
```
Comment (andrewphelpsj) [inline aspects/aspects.go:417]: should this be an "internal error:" ?

File: aspects/aspects.go [hunk 2]
```diff
@@ -503,6 +498,60 @@ func (a *Aspect) Set(databag DataBag, request string, value interface{}) error {
...
+		matches = append(matches, requestMatch{
+			storagePath: path,
+			suffixParts: suffixParts,
+			request:     rule.originalRequest,
+		})
+	}
+
+	if len(matches) == 0 {
+		return notFoundErrorFrom(a, "unset", request, "no matching write rule")
+	}
```
Comment (andrewphelpsj) [inline aspects/aspects.go:532]: Is it worth putting this section into a function shared between Aspect.Set and Aspect.Unset?

**Review by andrewphelpsj — APPROVED** [review]
Looks good, thanks!

File: aspects/aspects.go (hunk 2 above)
Comment (MiguelPires) [reply aspects/aspects.go:532]: That's a good idea, thanks

**Comment by pedronis** [comment]
@MiguelPires what happens if you send to set a prefix and the value is a nested map with null as leave values? I think with config option that's equivalent to unset (can you double check that)? what happens with aspects?

**Review by ZeyadYasser — APPROVED** [review]
LGTM, Thank you!

**Pull request merged**